### ISSUE TITLE
type validation in `v0.21.0`: `'false'` is not of type: `boolean`

### ISSUE THREAD

**Issue by swanderz** [dbt-msft/dbt-sqlserver#164]
in `v0.21.0` the strings `'true'` and `'false'` are no longer recognized as `boolean` types. Unfortunately there is no boolean type in TSQL, but there is a [bit type](https://docs.microsoft.com/en-us/sql/t-sql/data-types/bit-transact-sql?view=sql-server-ver15). @jtcohen6, any ideas for a workaround? I think we're blocked on releasing v0.21.0 until we figure something out. I tried integers and they didn't work as well.

https://github.com/dbt-msft/dbt-sqlserver/blob/4091a07a5bdb7b5bac521fc72d8da52fabcd6c16/dbt/include/sqlserver/macros/materializations/test.sql#L4-L7

```
'false' is not of type 'boolean'

Failed validating 'type' in schema['properties']['should_warn']:
    {'type': 'boolean'}

On instance['should_warn']:
    'false'
```

**Comment by jtcohen6** [comment]
Oh no! This was an unintended side effect of turning off agate type coercion for query results, an otherwise uncontroversially good thing.

Hmmm. It does sound like the [bit type](https://docs.microsoft.com/en-us/sql/t-sql/data-types/bit-transact-sql?view=sql-server-ver15) is what we're after, doesn't it? And we've got an `agate` table being returned here, and a [`convert_boolean-type`](https://github.com/dbt-msft/dbt-sqlserver/blob/4091a07a5bdb7b5bac521fc72d8da52fabcd6c16/dbt/adapters/sqlserver/impl.py#L31-L33) method defined in the adapter... I wonder if there isn't some way to muddle through this.

I'm thinking we might need to make this change in dbt-core, possibly [hereabouts](https://github.com/dbt-labs/dbt-core/blob/26625e9627261f12a278828a43defe750942cb4f/core/dbt/task/test.py#L31-L35):
```python
@dataclass
class TestResultData(dbtClassMixin):
    failures: int
    should_warn: bool
    should_error: bool
    
    @classmethod
    def validate(cls, data):
        if not isinstance(data.get('should_warn'), bool):
            data['should_warn'] = bool(data.get('should_warn'))
        if not isinstance(data.get('should_error'), bool):
            data['should_error'] = bool(data.get('should_error'))
        super().validate(data)
```

That would at least get you the ability to return `0` and `1`, which sounds like the T-SQL way.

Could I bother you to open a dbt-core issue for this?

**Comment by swanderz** [comment]
closed by #168
